src/lib/routing/response/index.ts

```diff
@@ -1,16 +1,24 @@
 import cookie from "cookie";
+import mitt, { Emitter } from "mitt";
 import { appendFn, setFn } from "../common";
 import { statusText } from "./status-codes";
 import { SurrogateKeys } from "./surrogate-keys";
 import { EHeaders } from "./headers";
 import { CookieOptions, EConfig } from "..";
 
+export type EResponseEvent = "finish";
+
+export type EResponseEvents = {
+  finish: Response;
+}
```

Comment:
I have a suggestion here: rather than sending the underlying `Response` object to the `'finish'` handler, I would consider sending the `EResponse` object, since most of the Expressly API surface (the route handlers) is about interacting with `ERequest`/`EResponse`. However, if you have a good reason for sending `Response` instead of `EResponse` then please feel free to disregard this suggestion (but please let me know what that is, I'm interested).

```suggestion
export type EResponseEvents = {
  finish: EResponse;
}
```

Reply:
Having the `Response` type here is intentional. This method was requested for debugging and post-request actions, and the parameter is intended to expose the `Response` object that was handed off to Compute after all expressly processing was executed. This is different from `EResponse` and the [few LOC](https://github.com/fastly/expressly/pull/52/files#diff-dba890aacb55bb4261ff3f61e32b34a0587b675e46340fde00ae69b101a62287R223-R232) above where the `finish` event is emitted under the hood will clarify why. I have made the call to send the `finalResponse: Response` as a parameter because this object isn't exposed anywhere else.

 `res` remains chainable. There's no need to inject it inside of a callback inside of its own method.

Reply:
Okay, thank you. That makes sense.